### PR TITLE
tests - test for #6416

### PR DESCRIPTION
**Pull request by commonism**
## Change Summary

adding a unit test for RootModel[List[RootModel[…]]]

see #6416

## Checklist

* [x] Unit tests for the changes exist
* [ ] Tests pass on CI and coverage remains at 100%
* [-] Documentation reflects the changes where applicable
* [x] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

File: changes/6416-commonism.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Add unit test for RootModel[List[RootModel[…]]]. See #6416.
```
Comment (hramezani) [inline changes/6416-commonism.md:1]: Don't need this file for changes like this

**Comment by commonism** [comment]
I'd say squash commits and it'll be fine?

File: tests/test_root_model.py [hunk 2]
```diff
@@ -518,6 +518,21 @@ class Model(RootModel):
...
+    class B(BaseModel):
+        type: Literal['b']
+        b: str
+
+    class D(RootModel[Annotated[Union[A, B], Field(discriminator='type')]]):
+        pass
+
+    RootModel[List[D]]
+
+
```
Comment (hramezani) [inline tests/test_root_model.py:535]:
```suggestion
    obj = LD.model_validate([{'type': 'a', 'a': 'a'}, {'type': 'b', 'b': 'b'}])
    assert obj.model_dump() == [{'type': 'a', 'a': 'a'}, {'type': 'b', 'b': 'b'}]


```

File: tests/test_root_model.py [hunk 3]
```diff
@@ -518,6 +518,24 @@ class Model(RootModel):
...
+        a: str
+
+    class B(BaseModel):
+        type: Literal['b']
+        b: str
+
+    class D(RootModel[Annotated[Union[A, B], Field(discriminator='type')]]):
+        pass
+
+    RootModel[List[D]]
```
Comment (hramezani) [inline tests/test_root_model.py:533]:
```suggestion
    LD = RootModel[List[D]]
```

**Review by hramezani — APPROVED** [review]

**Comment by hramezani** [comment]
Thanks @commonism

**Pull request merged**